### ISSUE TITLE
Replace global parameter `STRICT` with local argument `strict`

### ISSUE THREAD

**Issue by pietermarsman** [pdfminer/pdfminer.six#292]
https://github.com/pdfminer/pdfminer.six/blob/d6fd7e76b272e88dca67f7fabead397601ec6fce/pdfminer/settings.py#L1 

The global parameter `STRICT` from settings.py is used in a lot of different components of pdfminer. Having a global parameter implies that uses either want to do everything strict or non-strict. 

I think this is a false assumption and that we can improve the composability of pdfminer by making strict a argument of the classes and functions that use it. For example, this allows the parser to be strict and the interpreter to be non-strict.

**Comment by jstockwin** [comment]
Agreed, and I believe we discussed something similar to this on another issue about logging a warning if strict mode is off. I'm putting this card into "accepted" but feel free to move it into needs solution if you still think it needs discussion.

**Comment by pietermarsman** [comment]
This could also make strict mode more configurable like you suggested somewhere else :)

**Comment by pietermarsman** [comment]
I did a search on github for [the usage of](https://github.com/search?l=&p=1&q=%22settings.STRICT+%3D+True%22extension%3Apy&ref=advsearch&type=Code) `settings.STRICT = True` and it looks like almost no one is using it. The only usage I could find is here: https://github.com/lowatt/lowatt_pdfss/blob/1b1817cd428e9759a588868b9d6ae90cf8739957/pdfss/__init__.py#L108

Instead of refactoring to an argument, we can also remove the `settings.STRICT` option altogether. And just use `logger.warning()` every time something odds happens that can be fixed.
